src/dmd/dscope.d

```diff
@@ -708,6 +708,35 @@ struct Scope
         return null;
     }
 
+    /********************************************
+     * Find the lexically enclosing function (if any).
+     *
+     * This function skips through generated FuncDeclarations,
+     * e.g. rewritten foreach bodies.
+     *
+     * Returns: the function or null
+     */
+    inout(FuncDeclaration) getEnclosingFunction() inout
+    {
+        for (inout(Scope)* sc = &this; sc; sc = sc.enclosing)
+        {
+            if (!sc.func)
+                continue;
+
+            auto fd = cast(FuncDeclaration) this.func;
```

Comment:
It looks to me like this should be `sc.func` instead of `this.func`, or otherwise maybe the for loop is not necessary?

Reply:
Also, the cast is redundant. `sc.func` is already a `FuncDeclaration`

Reply:
> t looks to me like this should be sc.func instead of this.func, 

You're right, that was a typo.

> or otherwise maybe the for loop is not necessary?

Seems like it, I couldn't find a test case where the loop was necessary. Removed the loop.

Reply:
> Also, the cast is redundant. `sc.func` is already a `FuncDeclaration`

It's a `inout(FuncDeclaration)`